### PR TITLE
Publish snapshots of every master commit

### LINKED ISSUE #1254: Publish nightly snapshots
Nightly snapshots are good to have in general but especially for akka-http because we don't do any release candidates, so our releases are only tested by the project's automated tests and enthusiast that build from source.

Publishing the snapshots would help there, because we could people instantly point to a nightly with a fix which might help with catching regressions early on.

### PR DESCRIPTION
**Pull request by 2m**
To consume snapshots, the following resolver is to be used:

```scala
resolvers += Resolver.bintrayRepo("akka", "snapshots")
```

Fixes #1254

Supersedes #1744

### CODE REVIEW

File: project/Publish.scala [hunk 1]
```diff
@@ -30,7 +30,8 @@ object Publish extends AutoPlugin {
 
   override def projectSettings = Seq(
     bintrayOrganization := Some("akka"),
-    bintrayPackage := "com.typesafe.akka:akka-http_2.11"
+    bintrayPackage := "com.typesafe.akka:akka-http_2.11",
+    bintrayRepository := (if (isSnapshot.value) "snapshots" else "maven")
```
Comment (raboof) [inline project/Publish.scala]: This does risk doing the non-snapshot publish twice: when you create the latest commit on master and then tag it, travis will start 2 jobs, but both will see the tag and select 'maven' here. Perhaps just add a note to the release train to wait for this?

Comment (2m) [reply project/Publish.scala]: Yes. And such situation became especially common once we started committing release notes to documentation. After that PR is merged the tag is created usually very quickly afterwards, possibly making both builds release builds.

We have a [note about that in the Alpakka release train checklist](https://github.com/akka/alpakka/commit/2e295782a02dbdd0ff84b9299ea63040b360d25c#diff-5aa61b36f0395eb9d1fd1d14debe2e5cR39). Will add it to akka-http as well.

**Comment by 2m** [comment]
Release train issue template amended.

**Comment by jrudolph** [comment]
PLS BUILD

File: notes/release-train-issue-template.md [hunk 2]
```diff
@@ -38,6 +38,7 @@ Wind down PR queue. There has to be enough time after the last (non-trivial) PR
 ### Cutting the release
 
 - [ ] Make sure there are no stray staging repos on sonatype
+- [ ] Wait until [master build finished](https://travis-ci.org/akka/akka-http/builds/) after merging the release notes
```
Comment (jrudolph) [inline notes/release-train-issue-template.md]: Probably it's enough to wait for it to be started and has fetched the tags from git.

**Comment by jrudolph** [comment]
Thanks, @2m!

**Comment by jrudolph** [comment]
First run going on here: https://travis-ci.org/akka/akka-http/jobs/477749819

**Pull request merged**